### PR TITLE
DOC-10303 release/7.2: Adds URL functions to urlencode and urldecode

### PR DESCRIPTION
**Pull request by franrob-projects**
Adds URL functions to urlencode and urldecode

### CODE REVIEW

File: modules/n1ql/pages/n1ql-language-reference/stringfun.adoc [hunk 1]
```diff
@@ -1042,3 +1042,45 @@ SELECT UPPER("N1QL is awesome") as n1ql;
 }
 ----
 ====
+
+[#fn-str-upper]
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/stringfun.adoc]: Each function needs to have a unique anchor:

```suggestion
[#fn-str-urlencode]
```

File: modules/n1ql/pages/n1ql-language-reference/stringfun.adoc [hunk 2]
```diff
@@ -1042,3 +1042,45 @@ SELECT UPPER("N1QL is awesome") as n1ql;
...
+== URLENCODE(plain_string)
+
+=== Description
+Returns the input string encoded for use in a URL
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/stringfun.adoc]: Full stop at end

```suggestion
Returns the input string encoded for use in a URL.
```

File: modules/n1ql/pages/n1ql-language-reference/stringfun.adoc [hunk 3]
```diff
@@ -1042,3 +1042,45 @@ SELECT UPPER("N1QL is awesome") as n1ql;
...
+
+=== Arguments
+in_str:: A string, or any valid xref:n1ql-language-reference/index.adoc[expression] which evaluates to a string, that is the string to convert to upper case.
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/stringfun.adoc]: Needs to be consistent with syntax given in the function title. Also fix description:

```suggestion
plain_string:: A string, or any valid xref:n1ql-language-reference/index.adoc[expression] which evaluates to a string, that is the string to URL-encode.
```

File: modules/n1ql/pages/n1ql-language-reference/stringfun.adoc [hunk 4]
```diff
@@ -1042,3 +1042,45 @@ SELECT UPPER("N1QL is awesome") as n1ql;
...
+
+=== Return Value
+A string representing the input string converted to upper case.
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/stringfun.adoc]: Fix description. Also we need to cover the common edge cases &mdash; although I admit this page is not very good at that!

```suggestion
The input string, with any reserved characters replaced by percent encoding.

If the input argument is MISSING, returns MISSING.
If the input argument is non-string, returns NULL.
```

File: modules/n1ql/pages/n1ql-language-reference/stringfun.adoc [hunk 5]
```diff
@@ -1042,3 +1042,45 @@ SELECT UPPER("N1QL is awesome") as n1ql;
...
+
+=== Example
+====
+[source,text]
+----
+SELECT%20name%20FROM%20%60travel-sample%60.inventory.hotel%20LIMIT%201%3B
+----
+
+====
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/stringfun.adoc]: Okay, you've shown the _result_ of the function here, but we need to see a complete example, with input and output.

```suggestion
====
[source,sqlpp]
----
SELECT URLENCODE("SELECT name FROM `travel-sample`.inventory.hotel LIMIT 1;") AS encoded;
----

[source,json]
----
[
  {
    "encoded": SELECT%20name%20FROM%20%60travel-sample%60.inventory.hotel%20LIMIT%201%3B
  }
]
----
====
```

File: modules/n1ql/pages/n1ql-language-reference/stringfun.adoc [hunk 6]
```diff
@@ -1042,3 +1042,45 @@ SELECT UPPER("N1QL is awesome") as n1ql;
...
+
+[#fn-str-upper]
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/stringfun.adoc]: Each function needs to have a unique anchor:

```suggestion
[#fn-str-urldecode]
```

File: modules/n1ql/pages/n1ql-language-reference/stringfun.adoc [hunk 7]
```diff
@@ -1042,3 +1042,45 @@ SELECT UPPER("N1QL is awesome") as n1ql;
...
+== URLDECODE(encoded_string)
+
+=== Description
+Decodes the URL-encoded input string and returns the resulting plain string
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/stringfun.adoc]: Full stop at end:

```suggestion
Decodes the URL-encoded input string and returns the resulting plain string.
```

File: modules/n1ql/pages/n1ql-language-reference/stringfun.adoc [hunk 8]
```diff
@@ -1042,3 +1042,45 @@ SELECT UPPER("N1QL is awesome") as n1ql;
...
+
+=== Arguments
+in_str:: A string, or any valid xref:n1ql-language-reference/index.adoc[expression] which evaluates to a string, that is the string to convert to upper case.
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/stringfun.adoc]: Needs to be consistent with syntax given in the function title. Also fix description:

```suggestion
encoded_string:: A string, or any valid xref:n1ql-language-reference/index.adoc[expression] which evaluates to a string, that is the string to URL-decode.
```

File: modules/n1ql/pages/n1ql-language-reference/stringfun.adoc [hunk 9]
```diff
@@ -1042,3 +1042,45 @@ SELECT UPPER("N1QL is awesome") as n1ql;
...
+
+=== Return Value
+A string representing the input string converted to upper case.
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/stringfun.adoc]: Again, fix description, and cover the common edge cases:

```suggestion
The input string, with any percent encoding replaced by reserved characters.

If the input argument is MISSING, returns MISSING.
If the input argument is non-string, or if the argument is a string containing a single percent character `%`, returns NULL.
```

File: modules/n1ql/pages/n1ql-language-reference/stringfun.adoc [hunk 10]
```diff
@@ -1042,3 +1042,45 @@ SELECT UPPER("N1QL is awesome") as n1ql;
...
+
+=== Example
+====
+[source,text]
+----
+SELECT%20name%20FROM%20%60travel-sample%60.inventory.hotel%20LIMIT%201%3B”) AS decode;
+SELECT name FROM travel-sample.inventory.hotel LIMIT 1;
+----
+====
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/stringfun.adoc]: Again, need to show complete example, with input and output.

```suggestion
====
[source,sqlpp]
----
SELECT URLDECODE("SELECT%20name%20FROM%20%60travel-sample%60.inventory.hotel%20LIMIT%201%3B") AS decoded;
----

[source,json]
----
[
  {
    "decoded": "SELECT name FROM `travel-sample`.inventory.hotel LIMIT 1;"
  }
]
----
====
```

**Review by simon-dew — CHANGES_REQUESTED** [review]
Thanks @franrob-projects ! Got some comments for you. You should just be able to commit each suggestion.

File: modules/n1ql/pages/n1ql-language-reference/stringfun.adoc (hunk 2 above)
Comment (franrob-projects) [reply modules/n1ql/pages/n1ql-language-reference/stringfun.adoc]: I did wander about that full stop, I just copied what was there. Lesson for me — follow my intuition  💪

**Pull request merged**